core/commands/add.go

```diff
@@ -125,17 +103,12 @@ remains to be implemented.
 		outChan := make(chan interface{}, 8)
 		res.SetOutput((<-chan interface{})(outChan))
 
-		fileAdder := adder{
-			ctx:      req.Context(),
-			node:     n,
-			editor:   e,
-			out:      outChan,
-			chunker:  chunker,
-			progress: progress,
-			hidden:   hidden,
-			trickle:  trickle,
-			wrap:     wrap,
-		}
+		fileAdder := coreunix.NewAdder(req.Context(), n, outChan)
```

Comment:
does the Adder work just fine without specifying all the below parameters? (i.e. does it have sane defaults?)